contracts/cw1-subkeys/src/state.rs

```diff
@@ -9,19 +9,19 @@ use crate::balance::Balance;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema, Default)]
 pub struct Allowance {
-    balance: Balance,
-    expires: Expiration,
+    pub balance: Balance,
+    pub expires: Expiration,
 }
```

Comment:
Good call, these should all be public members

Reply:
I did it just for simplicity (avoiding getters / setters). I could implement those if you want, but I think this is clear, and good enough.

Reply:
It was a mistake by me. Any serializable object in cosmwasm code has all public members. They are just "dumb" data structures

Reply:
I like the approach.